### DISCUSSION TITLE
y-a-t-il un un emplacement microbit

### DISCUSSION THREAD

**Discussion by didier-fe** [micropython/micropython#13072] [General]
bonjour

je me suis mis à la microbit pour faire comme des amis. tous développent avec makecode. je désire passer en micropython.
et là je galère un peu car en plus de la maitrise du langage, il faut trouver les bibliothèques pour chaque élément rajouté. et là je suis un peu perdu. en cemoment j'en suis au i2clcd1602.

je suis arrivé ici en passant par l'ancien forum https://forum.micropython.org/viewforum.php?f=17 qui renvoie ici sur github. 
l'ancien forum possédait un onglet pour microbit.
répondez vous  pour microbit ? et dans ce cas où est-ce le mieux pour poser des questions ?
merci de votre compréhension

**Comment by rkompass** [comment]
hello @didier-fe 

Many of us are not fluent in french. So I suggest you translate your question with e.g.  [deepl](https://www.deepl.com/translator), which does very good translations, by he way. Which I did here:

Bonjour @didier-fe 

Beaucoup d'entre nous ne parlent pas couramment le français. Je vous suggère donc de traduire votre question avec, par exemple, [deepl](https://www.deepl.com/translator), qui fait de très bonnes traductions, soit dit en passant.


> Hello
> 
> i started using microbit to follow in the footsteps of my friends. they all develop with makecode. i want to switch to micropython.
> and now i'm having a hard time because in addition to mastering the language, i have to find the libraries for each added element. and now i'm a bit lost. at the moment i'm at i2clcd1602.
> 
> I got here via the old https://forum.micropython.org/viewforum.php?f=17 forum, which links here to github.
> the old forum had a tab for microbit.
> do you answer for microbit? and if so, where is the best place to ask questions?
> thank you for your understanding.

**Reply by rkompass** [reply]
Now in English: 
I did reworks of the i2c character LCD driver. What is your LCD interface board (the backpack board in the LCD display that exposes the I2C lines SDA and SCL)?
Or similar question: What kind of display do you have? This is important, as there are some differences in how the different versions of these boards have to be controlled.

**Reply by didier-fe** [reply]
Thank you for responding and sorry for my French.
with makecode it's quite easy, but I would like to learn to program in pyhton and it's difficult to find for microbit.

I have screen like https://www.gotronic.fr/art-afficheur-2x16-caracteres-lcd1602-26087.htm with https://www.gotronic.fr/art-backpack-i2c-pour-lcd-me033-37949.htm. the pictures look like what I have.

and i try to program ir remote https://www.gotronic.fr/art-telecommande-ir-irc01-19568.htm, but i need the IR program. i don't know what the name of ir card

thank for your help

**Reply by rkompass** [reply]
The LCD + backpack seem to closely match those that I did [this library](https://github.com/rkompass/Lcd_I2c8574) for. 
You have good chances that even the default settings (i2c address 0x27 and dim (16,2)) will work unchanged.

There are extensive instructions there that you may translate with deepl. Good luck.

**Reply by rkompass** [reply]
For the IR you need [this receiver](https://www.gotronic.fr/art-module-ir-38-khz-tsop34838-2280.htm).
For using it with Micropython I recommend the [library](https://github.com/peterhinch/micropython_ir/blob/master/RECEIVER.md) by @peterhinch. He also has extensive instructions there.

**Reply by didier-fe** [reply]
I thank you very much for your help. helping a beginner is not easy. 
well documented work, thank you.
Sorry for my delay in responding, very busy this week.

it remains complex. in makecode extensions + 4 blocks : connect lcd at I2C at 39, connect IR receiver at pin 0and decode nec, show IR datagram, 1lcd1602 show IR datagram.
my friends ask me why i don't stay at makecode, but i think it's worth it to learn python.

This seems optimized for pico and esp32 because I thought I saw a test to identify the card. I use BBC microbit

I don't know this command, I suppose for linux ?$ mpremote mip install "github:peterhinch/micropython_ir/ir_rx"
I will install the 8 -.py files in my project directory except .json (I don't know this extension) and try to imput them in my main program.

Thank you for your help. I'm trying this weekend

**Reply by rkompass** [reply]
Many here don't like it, but I still enjoy it's simplicity: [Thonny](https://thonny.org) . You may activate the File system views on your PC and on the Micropython Device (Micropbit) and simply copy files to and from by right mouse click (see e.g. [here](https://www.youtube.com/watch?v=CPkzcNIVqPQ)). So I didn't use mpremote so far (which is bad, but I didn't have to). 
For installation I simply copy the scripts to the file system of the boards.
